File: 20221012_Wednesday/20221012.rb
# Leetcode: 976. Largest Perimeter Triangle.
# https://leetcode.com/problems/largest-perimeter-triangle/
# @param {Integer[]} nums
# @return {Integer}
def largest_perimeter(nums)
  n = nums.sort.reverse
  i = 0
  while i + 2 < n.size
    if n[i] <= n[i+1] + n[i+2]
      return n[i] + n[i+1] + n[i+2]
    end
    i += 1
  end
  return 0
end

File: 20221012_Wednesday/20221012.go
// Leetcode: 976. Largest Perimeter Triangle.
// https://leetcode.com/problems/largest-perimeter-triangle/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 65 ms, faster than 52.00% of Go online submissions for Largest Perimeter Triangle.
// Memory Usage: 7 MB, less than 39.00% of Go online submissions for Largest Perimeter Triangle.
// 2022.10.12 Daily Challenge.
import "sort"
import "fmt"
func largestPerimeter(nums []int) int {
  sort.Ints(nums)
  for i:=len(nums)-1; i > 1; i-=1{
    if nums[i] < nums[i-1] + nums[i-2]{
      return nums[i] + nums[i-1] + nums[i-2]
    }
  }
  return 0
}
